bom validation needs to consider the schemaVersion

Using a schemaVersion property override of "1.1" in the current 1.2.x plugin versions generates a bom correctly but fails the validation step. Looks like this line needs to provide the schemaVersion param as well: https://github.com/CycloneDX/cyclonedx-gradle-plugin/blob/cyclonedx-gradle-plugin-1.2.1/src/main/java/org/cyclonedx/gradle/CycloneDxTask.java#L383.

XmlParser uses the default schema version (1.2) resulting in error message `The BOM does not conform to the CycloneDX BOM standard`

@stevespringett  yes it seems like this is still an issue with 1.3.0

## Environment:
Gradle 7.0
Build time:   2021-04-09 22:27:31 UTC
Revision:     d5661e3f0e07a8caff705f1badf79fb5df8022c4

Kotlin:       1.4.31
Groovy:       3.0.7
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          1.8.0_282 (AdoptOpenJDK 25.282-b08)

## Command Output:
```
➜ gradle cyclonedxBom -info
Initialized native services in: /Users/arturo.alonso/.gradle/native
The client will now receive all logging from the daemon

(pid: 71940). The daemon log file: /Users/arturo.alonso/.gradle/daemon/7.0/daemon-71940.out.log
Starting 4th build in daemon [uptime: 1 mins 50.613 secs, performance: 99%, non-heap usage: 23% of 256 MiB]
Using 12 worker leases.
Now considering [/private/var/folders/xq/3yjf7y6d6kx4jfjjfyfn7hkr0000gq/T/junit2848716798378917290] as hierarchies to watch
Not watching anything anymore
Watching the file system is enabled if available
Starting Build
Settings evaluated using settings file '/private/var/folders/xq/3yjf7y6d6kx4jfjjfyfn7hkr0000gq/T/junit2848716798378917290/settings.gradle'.
Projects loaded. Root project using build file '/private/var/folders/xq/3yjf7y6d6kx4jfjjfyfn7hkr0000gq/T/junit2848716798378917290/build.gradle'.
Included projects: [root project 'hello-world']

> Configure project :
Evaluating root project 'hello-world' using build file '/private/var/folders/xq/3yjf7y6d6kx4jfjjfyfn7hkr0000gq/T/junit2848716798378917290/build.gradle'.
All projects evaluated.
Selected primary task 'cyclonedxBom' from project :
Tasks to be executed: [task ':cyclonedxBom']
Tasks that were excluded: []
:cyclonedxBom (Thread[Execution worker for ':',5,main]) started.

> Task :cyclonedxBom FAILED
Caching disabled for task ':cyclonedxBom' because:
  Build cache is disabled
Task ':cyclonedxBom' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
CycloneDX: Parameters
------------------------------------------------------------------------
schemaVersion          : VERSION_11
includeBomSerialNumber : true
------------------------------------------------------------------------
CycloneDX: Resolving Dependencies
CycloneDX: Creating BOM
CycloneDX: Writing BOM XML
CycloneDX: Validating BOM
:cyclonedxBom (Thread[Execution worker for ':',5,main]) completed. Took 0.042 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':cyclonedxBom'.
> The BOM does not conform to the CycloneDX BOM standard

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 594ms
1 actionable task: 1 executed
```
## Sample project:
[sample-project.zip](https://github.com/CycloneDX/cyclonedx-gradle-plugin/files/6395528/sample-project.zip)

--- a/src/main/java/org/cyclonedx/gradle/CycloneDxTask.java
+++ b/src/main/java/org/cyclonedx/gradle/CycloneDxTask.java
@@ -382,7 +382,7 @@ public class CycloneDxTask extends DefaultTask {
         getLogger().info(MESSAGE_VALIDATING_BOM);
         final Parser bomParser = new XmlParser();
         try {
-            if (!bomParser.isValid(bomFile)) {
+            if (!bomParser.isValid(bomFile, schemaVersion)) {
                 throw new GradleException(MESSAGE_VALIDATION_FAILURE);
             }
         } catch (Exception e) { // Changed to Exception.
@@ -400,7 +400,7 @@ public class CycloneDxTask extends DefaultTask {
         getLogger().info(MESSAGE_VALIDATING_BOM);
         final Parser bomParser = new JsonParser();
         try {
-            if (!bomParser.isValid(bomFile)) {
+            if (!bomParser.isValid(bomFile, schemaVersion)) {
                 throw new GradleException(MESSAGE_VALIDATION_FAILURE);
             }
         } catch (Exception e) { // Changed to Exception.

